### DISCUSSION TITLE
Access each 16-bit word of a 32-bit value with ModbusSimulatorContext

### DISCUSSION THREAD

**Discussion by haxj** [pymodbus-dev/pymodbus#1550] [Q&A] [answered]
Hi all,

I have some questions regarding the following scenario:

Suppose there's a uint32 value at address `addr`. From the client, I can access that 32-bit value by sending a Modbus request with `address = addr` and `count = 2`. Logically I think the client can also send 2 Modbus requests with `count = 1` and address set to `addr` and `addr + 1`  to access the high and low word separately.

My questions are:
- I understand that the Modbus specs doesn't officially support 32-bit values, however in the real world, is this scenario common?
- Do real Modbus servers/slaves usually support accessing each word of a 32-bit value separately like this?
- If yes, should the class `ModbusSimulatorContext` support this kind of access in the future? Currently `ModbusSimulatorContext` stores the whole 32-bit value at `addr` without caring about its high/low 16-bit words.

**Comment by janiversen** [comment, marked as answer]
it does support 32 bit values, if you look more in detail it is stored in 2 addresses (each 16bit).

and it does care about the big/little indian conversion. please remember the modbus protocol always use big indian.

**Reply by haxj** [reply]
Thanks for your confirmation. I'm using `ModbusSimulatorContext` but had some issues with this scenario, so I thought there's no support yet. But if it already does, then may be I'm using it wrong or there's a bug.

I'll do some more tests to pinpoint the issues.
